common/big_messaging_typesafe/src/main/scala/uk/ac/wellcome/bigmessaging/typesafe/VHS.scala

```diff
@@ -0,0 +1,124 @@
+package uk.ac.wellcome.bigmessaging.typesafe
+
+import java.util.UUID
+import scala.util.{Failure, Success, Try}
+import org.scanamo.{DynamoFormat, DynamoValue}
+import org.scanamo.error.DynamoReadError
+import org.scanamo.auto._
+import com.typesafe.config.Config
+import com.amazonaws.services.s3.AmazonS3
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB
+
+import uk.ac.wellcome.typesafe.config.builders.EnrichConfig._
+
+import uk.ac.wellcome.storage.store.s3.S3TypedStore
+import uk.ac.wellcome.storage.store.dynamo.DynamoHashStore
+import uk.ac.wellcome.storage.dynamo.DynamoConfig
+import uk.ac.wellcome.storage.store.s3.S3TypedStore
+import uk.ac.wellcome.storage.store.{
+  HybridIndexedStoreEntry,
+  HybridStoreWithMaxima,
+  Store,
+  TypedStore,
+  VersionedHybridStore
+}
+import uk.ac.wellcome.storage.{
+  Identified,
+  ObjectLocation,
+  ObjectLocationPrefix,
+  Version
+}
+import uk.ac.wellcome.storage.typesafe.{DynamoBuilder, S3Builder}
+import uk.ac.wellcome.storage.streaming.Codec
+import uk.ac.wellcome.storage.maxima.Maxima
+
+case class EmptyMetadata()
+
+trait GetLocation {
+  def getLocation(key: Version[String, Int]): Try[ObjectLocation]
+}
+
+class VHS[T](val hybridStore: VHSInternalStore[T])
+    extends VersionedHybridStore[
+      String,
+      Int,
+      ObjectLocation,
+      T,
+      EmptyMetadata
+    ](hybridStore)
+    with GetLocation {
+
+  def getLocation(key: Version[String, Int]): Try[ObjectLocation] =
+    hybridStore.indexedStore.get(key) match {
+      case Right(Identified(_, entry)) => Success(entry.typedStoreId)
+      case Left(error)                 => Failure(error.e)
+    }
+}
+
+class VHSInternalStore[T](
```

Comment:
What's the difference between a VHS and a VHSInternalStore?

Reply:
If you look at the defintion of [`VersionedHybridStore`](https://github.com/wellcometrust/scala-storage/blob/master/storage/src/main/scala/uk/ac/wellcome/storage/store/VersionedHybridStore.scala) in `scala-storage`, it contains another store that is passed in with the `hybridStore` parameter. These classes here maintain the same relationsip, i.e.

* `VersionedHybridStore` `~>` `VHS`
* `HybridStoreWithMaxima` `~>` `VHSInternalStore`